### ISSUE TITLE
OpenFileOnRemoteFrom should keep line numbers, or have an option to do so

### ISSUE THREAD

**Issue by jakub-g** [gitkraken/vscode-gitlens#3021]
# TL;DR 

I think `OpenFileOnRemoteFrom` should contain line numbers, similar to `OpenFileOnRemote`. In some % of cases, the result will be slightly incorrect line numbers, but this is better than having no line numbers at all.

# Problem

Currently when using `OpenFileOnRemote` the generated URL contains line number(s).

However `OpenFileOnRemoteFrom` (for choosing different branch name first), the link generated doesn't contain line numbers, on purpose, because they can be wrong:

https://github.com/gitkraken/vscode-gitlens/blob/a0a68919fee9905418f63c6c7409c68e22b8ebcd/src/commands/openFileOnRemote.ts#L107

**While this is technically true, I believe it's not helpful in many cases.**

# Context

Things may vary to other developers, but let me share my thought process.

Most of them time during development, I'm working in a `random_feature_branch`.

When I decide to open the file on remote UI, I can choose either `OpenFileOnRemote` and implicitly use `random_feature_branch`, or `OpenFileOnRemoteFrom` and use a different branch. In that latter case, 99% of time, I will use `main` branch.

Why do I want the latter?

- Most of the times when I open on remote, I actually want to share the URL with coworker.
  - Usually I want to discuss something _not relevant specifically to `random_feature_branch`, but a generic thing that came up._ 
  - If I wanted to discuss something specific to the diff I made in `random_feature_branch`, I'd rather create a pull request, and send a deep link to the pull request view.
 
**Note: Once I made the decision to use `main` branch, I just made a conscious decision that the link to another branch will be "better" than a link to my current branch:**

- I don't want to expose an "ugly" link with branch name
- Often times, `random_feature_branch` is not yet pushed to origin, so opening current branch won't work
- Even if `random_feature_branch` is up, once my PR is merged, it will be auto-deleted. I don't want to send links that will stop working in a few days.
- Often times, I will use `main` branch, and sometimes I will press `y` in GitHub UI to convert the link to permalink to a sha (I could do the same with current branch, but the sha generated in this way may disappear after rebase + garbage collection on the server; so not great).

**Most likely, since I made the decision to use a different branch, this means that the files are likely the same or very similar in the two branches. If I didn't think so, I'd rather generate a link to the current branch.**

**If the link turns out to be slightly incorrect**: it's easy to fix it, by clicking a few lines above or below in GitHub interface. 

Whereas if there's no line number in the URL, I need: to go back to my editor; check which line(s) I had selected; scroll to that line; and click to highlight it.

**Comment by sorenlouv** [comment]
Completely agree with this. 

@d13 Would it be possible to keep the current behaviour as default, but making it configurable so those who want to retain line numbers can do that? That seems like a nice compromise where everyone can have it their way.
